dnf5-plugins/automatic_plugin/transaction_callbacks_simple.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "transaction_callbacks_simple.hpp"
+
+#include <libdnf5/rpm/nevra.hpp>
+#include <libdnf5/rpm/transaction_callbacks.hpp>
+#include <libdnf5/transaction/transaction.hpp>
+
+#include <iostream>
+
+namespace dnf5 {
+
+void TransactionCallbacksSimple::transaction_start([[maybe_unused]] uint64_t total) {
+    output_stream << "  Prepare transaction" << std::endl;
+}
+
+
+void TransactionCallbacksSimple::install_start(
+    const libdnf5::rpm::TransactionItem & item, [[maybe_unused]] uint64_t total) {
+    switch (item.get_action()) {
+        case libdnf5::transaction::TransactionItemAction::UPGRADE:
+            output_stream << "  Upgrading ";
+            break;
+        case libdnf5::transaction::TransactionItemAction::DOWNGRADE:
+            output_stream << "  Downgrading ";
+            break;
+        case libdnf5::transaction::TransactionItemAction::REINSTALL:
+            output_stream << "  Reinstalling ";
+            break;
+        case libdnf5::transaction::TransactionItemAction::INSTALL:
+            output_stream << "  Installing ";
+            break;
+        case libdnf5::transaction::TransactionItemAction::REMOVE:
+        case libdnf5::transaction::TransactionItemAction::REPLACED:
+            break;
+        case libdnf5::transaction::TransactionItemAction::REASON_CHANGE:
+        case libdnf5::transaction::TransactionItemAction::ENABLE:
+        case libdnf5::transaction::TransactionItemAction::DISABLE:
+        case libdnf5::transaction::TransactionItemAction::RESET:
+            throw std::logic_error(fmt::format(
```

Comment:
Do I understand it correctly that when the logic_error is triggered here it will terminate RPM transaction? The consequence would be a broken system or did I overlooked something?

Reply:
I've added a commit which removes throwing exception in transaction callbacks.